Domain Site Redirect: Use shoppingCartManager to add products in SiteRedirectStep

sirbrillig opened the pull request:
#### Changes proposed in this Pull Request

As part of removing `CartStore` (see https://github.com/Automattic/wp-calypso/issues/24019), this PR replaces the code in `SiteRedirectStep` that adds a redirect product to the cart with functions from `@automattic/shopping-cart`.

![site-redirect-step](https://user-images.githubusercontent.com/2036909/108778622-9ea00080-7533-11eb-984b-23bf19c7640c.gif)


#### Testing instructions

- Visit the site-redirect form at a URL like http://calypso.localhost:3000/domains/add/site-redirect/example.com
- Write a domain name in the form field and press "Go".
- Verify that you are redirected to checkout and that a "Site redirect" product is in your cart.

sirbrillig committed:
Use shoppingCartManager to add products in SiteRedirectStep

sirbrillig committed:
Remove getting productsList from state as we already have it

sirbrillig committed:
Remove unnecessary `cart` prop on SiteRedirect as we already have it

sirbrillig committed:
Remove unnecessary CartData wrapper on SiteRedirect

sirbrillig committed:
Wrap SiteRedirect in a CalypsoShoppingCartProvider

klimeryk reviewed (commented):
Tested and works - with the exception that there's a significant delay between clicking the "Add" button and anything happening. I know it's because of the async call to the shopping cart, etc., but to the user (or a CEO testing these flows... ;)), it could appear like you need to click a few times to get it to "work". Would it be possible to add a `busy` state to the button while we're waiting for the shopping cart to refresh/reload?

sirbrillig committed:
Make button busy while form is submitting

sirbrillig committed:
Also disable button when submitting

sirbrillig commented:
Good thinking. I added a busy state and disabled the button (I always forget that the former does not imply the latter). Updated the description to include a gif.

sirbrillig commented:
Failing tests seem to be unrelated.

nbloomf commented:
Testing works as expected, and the busy/disabled button also works. Using the X to leave checkout goes back to the Site Redirect page.
